lib/spack/spack/spec_parser.py

```diff
@@ -268,6 +291,9 @@ def add_dependency(dep, **edge_properties):
             except spack.error.SpecError as e:
                 raise SpecParsingError(str(e), self.ctx.current_token, self.literal_str) from e
 
+        # Get toolchain information outside of loop
+        toolchains = spack.config.CONFIG.get("toolchains", {})
```

Comment:
(minor) It would be good if we could avoid this here

Reply:
I think we want the toolchain parts of parser to be in a separate file (e.g. `toolchain.py`), as I suspect it'll grow.  e.g., we likely want something like this to clean up `spack spec` output (maybe not for 1.0 but eventually) and that can go in the dedicated toolchain module.